numpy/ma/tests/test_old_ma.py

```diff
@@ -289,20 +297,22 @@ def test_testPut(self):
         d = arange(5)
         n = [0, 0, 0, 1, 1]
         m = make_mask(n)
+        m2 = m.copy()
         x = array(d, mask=m)
         self.assertTrue(x[3] is masked)
         self.assertTrue(x[4] is masked)
         x[[1, 4]] = [10, 40]
-        self.assertTrue(x.mask is not m)
```

Comment:
The changes I am making here are similar to the ones we have seen in the last two. It is hard to know the original author's intention without discussing with him about it.

Regardless, we don't copy the data by default and I think that what we do with the mask should mimic what we do with the data. Otherwise, we find changes to the mask **are not** back propagated when we find changes to the data **are** back propagated. This sort of inconsistence is a nightmare for the end user and really seems unjustified.

Unfortunately, the masked array behavior hasn't been as well documented as other parts of NumPy. If there is somewhere you would like me to add some documentation about this please let me know.